core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/pieChart/PieChartOpFinalExec.java

```diff
@@ -49,23 +51,38 @@ public String getParam(String query) {
     @Override
     public scala.collection.Iterator<Tuple> processTexeraTuple(Either<Tuple, InputExhausted> tuple, LinkIdentity input) {
         if (tuple.isLeft()) {
-            sum += tuple.left().get().getDouble(1);
+            if (noDataCol) {
+                sum += tuple.left().get().getInt(1);
+            } else {
+                sum += tuple.left().get().getDouble(1);
+            }
             tempList.add(tuple.left().get());
             if (resultSchema == null) resultSchema = tuple.left().get().getSchema();
             return JavaConverters.asScalaIterator(Collections.emptyIterator());
         } else {
             // sort all tuples in descending order
             tempList.sort((left, right) -> {
-                double leftValue = left.getDouble(1);
-                double rightValue = right.getDouble(1);
+                double leftValue;
+                double rightValue;
+                if (noDataCol) {
+                    leftValue = left.getInt(1);
+                    rightValue = right.getInt(1);
+                } else {
+                    leftValue = left.getDouble(1);
+                    rightValue = right.getDouble(1);
+                }
                 return Double.compare(rightValue, leftValue);
             });
 
             // process the sorted rows, if the cumulative sum is greater than ratio * sum.
             // stop adding tuples, add new row called "Other" instead.
             double total = 0.0;
             for (Tuple t : tempList) {
-                total += t.getDouble(1);
+                if (noDataCol) {
+                    total += t.getInt(1);
+                } else {
+                    total += t.getDouble(1);
+                }
                 resultList.add(t);
                 if (total / sum > pruneRatio) {
                     double otherDataField = sum - total;
```

Comment:
In line 89, I see a bug here: when there is no data column and the prune ratio is less than 1 (where this line will be executed), there will be type mismatch for `count`.

Reply:
fixed